src/KubeOps/Operator/Builder/OperatorBuilder.cs

```diff
@@ -123,8 +121,8 @@ internal IOperatorBuilder AddOperatorBase(OperatorSettings settings)
                 },
                 DateFormatString = "yyyy'-'MM'-'dd'T'HH':'mm':'ss.ffffffK",
             };
-            Services.AddTransient(_ => jsonSettings);
-            JsonConvert.DefaultSettings = () => jsonSettings;
+
+            Services.AddSingleton(settings);
```

Comment:
correlates with the comment below.
```suggestion
            Services.AddSingleton(settings);
            Services.AddSingleton(new JsonSerializerSettings { ... });
```

and then inject the settings when needed (instead of operator settings and set a particular field to `null!`)